pkg/pgmodel/new_migrate.go

```diff
@@ -0,0 +1,137 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package pgmodel
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/blang/semver/v4"
+	"github.com/jackc/pgx/v4"
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/migrations"
+	"github.com/timescale/promscale/pkg/pgmodel/common/errors"
+	"github.com/timescale/promscale/pkg/pgmodel/common/extension"
+	"github.com/timescale/promscale/pkg/util"
+)
+
+var (
+	MigrationLockError = fmt.Errorf("Could not acquire migration lock. Ensure there are no other connectors running and try again.")
+	migrateMutex       = &sync.Mutex{}
+)
+
+func doesSchemaMigrationTableExist(db *pgx.Conn) (exists bool, err error) {
+	const stmt = "SELECT count(*) FILTER (WHERE tablename = 'prom_schema_migrations') > 0 FROM pg_tables"
+	err = db.QueryRow(
+		context.Background(),
+		stmt,
+	).Scan(&exists)
+	return
+}
+
+func removeOldExtensionIfExists(db *pgx.Conn) error {
+	const (
+		// transition is the first version of the extension that does the
+		// migrations the new way (i.e. in the extension rather than from promscale connector)
+		transition = "0.5.0"
+		stmt       = "DROP EXTENSION IF EXISTS promscale" // TODO to cascade or not to cascade?
+	)
+
+	installedVersion, installed, err := extension.FetchInstalledExtensionVersion(db, "promscale")
+	if err != nil {
+		return fmt.Errorf("error fetching extension version while dropping old extension: %w", err)
+	}
+
+	if installed && installedVersion.LT(semver.MustParse(transition)) {
+		_, err := db.Exec(
+			context.Background(),
+			stmt,
+		)
+		if err != nil {
+			return fmt.Errorf("error dropping old extension: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func installExtensionAllBalls(db *pgx.Conn) error {
+	const stmt = "CREATE EXTENSION promscale SCHEMA _prom_ext VERSION '0.0.0'"
+	_, err := db.Exec(
+		context.Background(),
+		stmt,
+	)
+	return fmt.Errorf("error installing Promscale extension at version 0.0.0: %w", err)
```

Comment:
This line is incorrect. Should be:

```golang
	if err != nil {
		return fmt.Errorf("error installing Promscale extension at version 0.0.0: %w", err)
	}
	return nil
```

Reply:
Good catch! thanks. addressed in https://github.com/timescale/promscale/pull/1132/commits/49d90e1e1024c94ab24fb3024b5bf273d63539bf